.grit/patterns/java/assignment_comparison.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Assignment Comparison
+tags: [java, good-practice]
+---
+
+Consolidate assignment-based conditionals `($var = true/false)` and equality checks `($var == true/$var)` into a single simplified conditional `(if($var))` to enhance code clarity and prevent unintended variable modifications.
+
+```grit
+language java
+
+or {
+    `if($var = true) { $body }`, 
+    `if($var = false) { $body }`,
+    `if($var == true) { $body }`,
+    `if($var == $var) { $body }`
```

Comment:
This identity check is always true and could be removed entirely, so you end up with just `$body`.